src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedSimpleProgramEntryPointSymbol.cs

```diff
@@ -33,20 +33,29 @@ internal SynthesizedSimpleProgramEntryPointSymbol(SimpleProgramNamedTypeSymbol c
             _declaration = declaration;
 
             bool hasAwait = declaration.HasAwaitExpressions;
+            bool hasReturnWithExpression = declaration.HasReturnWithExpression;
 
-            if (hasAwait)
+            switch (hasAwait, hasReturnWithExpression)
             {
-                _returnType = Binder.GetWellKnownType(containingType.DeclaringCompilation, WellKnownType.System_Threading_Tasks_Task, diagnostics, NoLocation.Singleton);
-            }
-            else
-            {
-                _returnType = Binder.GetSpecialType(containingType.DeclaringCompilation, SpecialType.System_Void, NoLocation.Singleton, diagnostics);
+                case (true, false):
+                    _returnType = Binder.GetWellKnownType(containingType.DeclaringCompilation, WellKnownType.System_Threading_Tasks_Task, diagnostics, NoLocation.Singleton);
+                    break;
+                case (false, false):
+                    _returnType = Binder.GetSpecialType(containingType.DeclaringCompilation, SpecialType.System_Void, NoLocation.Singleton, diagnostics);
+                    break;
+                case (true, true):
+                    _returnType = Binder.GetWellKnownType(containingType.DeclaringCompilation, WellKnownType.System_Threading_Tasks_Task_T, diagnostics, NoLocation.Singleton).
```

Comment:
>NoLocation.Singleton [](start = 153, length = 20)

Should this be the location of the first `return` expression?

Reply:
> Should this be the location of the first return expression?

I do not believe this edge case is worth the effort locating the one return statement with expression.

---
In reply to: [410423382](https://github.com/dotnet/roslyn/pull/43451#discussion_r410423382) [](ancestors = 410423382)